### PR TITLE
Issue #13038: handled inner class method definition in VariableDeclar…

### PR DESCRIPTION
**Pull request by smita1078**
#13038 

Extends #16026 
WIP
This PR updates `VariableDeclarationUsageDistanceCheck` to properly handle inner scopes such as:

- Inner classes declared inside methods
- Lambdas
- Anonymous class overrides
- Added multiple Tests

When `validateBetweenScopes = true`, the check now treats these nested declarations as **independent scopes**, and does **not flag** variables whose usages lie fully within those inner scopes.

This prevents false positives when a variable is declared before an inner class or lambda but only used inside it — aligning the behavior with expectations documented in issue #13038.

### CODE REVIEW

**Comment by smita1078** [comment]
Hi  @romani ,

I've tried to fix issue #13038.

The idea I followed is: when `validateBetweenScopes = true`, the check should **not count distance across nested scopes** like inner classes, lambdas, or anonymous classes declared inside methods. So, variables declared in a method and used only inside such nested scopes won't trigger a violation.

I've added/updated test cases for those scenarios ,but I'm not fully sure if this direction is correct.
Please review and let me know if the approach makes sense ?

**Comment by romani** [comment]
https://checkstyle.org/checks/coding/variabledeclarationusagedistance.html#Example5-code

`<property name="validateBetweenScopes" value="true"/>`

is supposed to be used for code like:
```
  public void foo2() {
    int a;          // ok, distance = 2
    int b;          // ok, distance = 3
    // violation below, 'variable 'count' declaration and its first usage is 4.'
    int count = 0;

    {   // this is inner scope start
      System.out.println("Inside inner scope");
      a = 1;
      b = 2;
      count++;
    } // this is inner scope end
  }
```

where `{ }` is used not for method/classes bodies.

it should not be applicable for classes/methods/if/........ , it is should be applicable for only for curly braces that stand-alone. Compiler allows it and user sometime like  such structure.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/InputVariableDeclarationUsageDistanceGeneral3.java [hunk 1]
```diff
@@ -25,7 +25,7 @@ <T> void xx(List<T> m){}
     }
 
     public void method2() {
-        Integer a = 5;
+        Integer a = 5; // violation 'Distance .* is 2.'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/InputVariableDeclarationUsageDistanceGeneral3.java]: distance should be 1, as `class AClass extends Parent {` is class start not a scope, variable cannot be moved to any better place.
So no violation should be here

**Review by romani — CHANGES_REQUESTED** [review]
itms:

**Comment by romani** [comment]
@smita1078 , is there a chance to finish this PR ?

**Comment by smita1078** [comment]
> @smita1078 , is there a chance to finish this PR ?

Hi @romani ,
I did explore multiple logic updates and test scenarios to ensure correct handling of validateBetweenScopes=true, but unfortunately, I’m still unable to fully figure out the correct handling for all edge cases. At this point, I’m not able to make further progress on this PR.

**Comment by romani** [comment]
It is completely ok to abandon PR. 
Not all is easy and sometimes need second iterations or some break from task to find solution.

We would deeply appreciate if you can help us with some other issues.

**Comment by smita1078** [comment]
> It is completely ok to abandon PR. Not all is easy and sometimes need second iterations or some break from task to find solution.
> 
> We would deeply appreciate if you can help us with some other issues.

Thank you so much for understanding.
I’ll look into some other issues and continue contributing where I can.
